searchlib/src/vespa/searchlib/queryeval/weighted_set_term_blueprint.cpp

```diff
@@ -96,11 +96,15 @@ SearchIterator::UP
 WeightedSetTermBlueprint::createLeafSearch(const fef::TermFieldMatchDataArray &tfmda, bool) const
 {
     assert(tfmda.size() == 1);
+    if ((_terms.size() == 1) && tfmda[0]->isNotNeeded() && _terms[0]->isLeaf()) {
```

Comment:
consider doing this as an actual optimization

Reply:
Turned out that was not possible due to mismatch of handle vs matchdata mismatch.